src/IOCapture.jl

```diff
@@ -105,7 +111,21 @@ function capture(f; rethrow::Type=Any, color::Bool=false)
     # pipe to `output` in order to avoid the buffer filling up and stalling write() calls in
     # user code.
     output = IOBuffer()
-    buffer_redirect_task = @async write(output, pipe)
+    if pass_through
+        bufsize = 128
+        buffer = Vector{UInt8}(undef, bufsize)
+        buffer_redirect_task = @async begin
+            while true
+                nbytes = readbytes!(pipe, buffer, bufsize)
+                data = view(buffer, 1:nbytes)
+                write(output, data)
+                write(default_stdout, data)
+                isopen(pipe) || break
```

Comment:
There might be a race condition here:

1. `buffer_redirect_task` reads from pipe
2. main task writes to pipe and closes it
3. `buffer_redirect_task` ends the loop

Possible result: the data the main task wrote is lost

Also: what is the best condition for ending the loop? An alternative might be `nbytes == 0`, although I'm not sure if that could get triggered accidentally if the main task doesn't write enough data to the pipe

Reply:
Should be solved via a suggestion from @Seelengrab on Slack: use `!eof(pipe)` instead of `isopen(pipe)`.